app/src/routes/api/v2/area.router.js

```diff
@@ -134,7 +134,17 @@ class AreaRouterV2 {
 
         const areas = await AreaModel.paginate(filter, { page, limit, sort: filteredSort });
 
-        await Promise.all(areas.docs.map((el) => SubscriptionService.mergeSubscriptionSpecificProps(el, ctx.request.headers['x-api-key'])));
+        const delay = (ms) => new Promise((resolve) => { setTimeout(resolve, ms); });
```

Comment:
I think this looks good, my only concern is could the increasing delay potentially lead to the request timing out over 150 areas? Like in this case, won't be a 7.5 seconds to process 150 areas?

Is there an easy way in which you could batch the request to subscriptions for all areas at once?

Feel free to merge if you think this approach is best and that shouldn't be an issue.

Reply:
Batching is the best fix for sure. I was trying to keep to something simple just to get them working again.
Yes, 7.5 seconds. Our default is a page size of 300 so 15 seconds in all.

GFW ignores the pagination I found out.

I'll knock it down to 25 ms to give us more wiggle room. I think they're really close to succeeding but during busy times it kills the server.